libcalico-go/lib/ipam/ipam.go

```diff
@@ -370,7 +386,35 @@ func (c ipamClient) prepareAffinityBlocksForHost(ctx context.Context, requestedP
 	}
 
 	// Determine the correct set of IP pools to use for this request.
-	poolsSelectingNode, allPools, err := c.determinePools(ctx, requestedPools, version, *v3n, maxPrefixLen)
+	// For LoadBalancer use, we don't have namespace context, so use empty values
```

Comment:
```suggestion
	// For some IPs (e.g., tunnel addresses), we don't have namespace context, so use empty values
```

Reply:
I think it would be cleaner to simply pass `nil` and to have `determinePools` handle that?

Looks like `SelectsNamespace` already handles a nil input, so I think we can remove this whole section and simply use `namespace` instead of `namespaceObj` here?